python/pyspark/ml/classification.py

```diff
@@ -3151,7 +3151,7 @@ def func(predictions):
                     predArray.append(x)
                 return Vectors.dense(predArray)
 
-            rawPredictionUDF = udf(func)
+            rawPredictionUDF = udf(func, VectorUDT())
             aggregatedDataset = aggregatedDataset.withColumn(
                 self.getRawPredictionCol(), rawPredictionUDF(aggregatedDataset[accColName]))
```

Comment:
```suggestion
            aggregatedDataset = aggregatedDataset.withColumn(
                self.getRawPredictionCol(), aggregatedDataset[accColName].cast(VectorUDT()))
```

Reply:
We should use `udf(func, VectorUDT())`.

Reply:
@HyukjinKwon 
I want to know if no udf return type specified, how does the return type inferring work ? Check all rows udf return type ?
The master code failed in some cases and the return column type in schema become "String".

Reply:
Got it.